src/DynamoPackages/PackageLoader.cs

```diff
@@ -418,16 +418,45 @@ public Package ScanPackageDirectory(string directory, bool checkCertificates)
                 // prevent loading unsigned packages if the certificates are required on package dlls
                 if (checkCertificates)
                 {
-                    CheckPackageNodeLibraryCertificates(directory, discoveredPkg);
+                    CheckPackageNodeLibraryCertificates(directory, discoveredPackage);
                 }
 
-                // prevent duplicates
-                if (LocalPackages.All(pkg => pkg.Name != discoveredPkg.Name))
+                var discoveredVersion = CheckAndGetPackageVersion(discoveredPackage.VersionName, discoveredPackage.Name, discoveredPackage.RootDirectory);
+
+                var existingPackage = LocalPackages.FirstOrDefault(package => package.Name == discoveredPackage.Name);
```

Comment:
@sm6srw will this necessarily discover ALL conflicts, what if there are 3 versions of the package?

Reply:
also - isn't `LocalPackages` only populated once a package is actually loaded?

Reply:
Yes, it will find all conflicts as it make sure that a package can only exist once in LocalPackages. A package will be added the first time and then replaced with a new version or ignored if older going forward.

Reply:
Lock at the `LoadAll` method. It first scan all package directories and populate `LocalPackages`. It then uses `LocalPackages` for loading all packages.